File: RB101/study_guide/live_assesment/small_problems/easy_2/sum_product.rb
def arrows(string)
  puts ">> " + string
end

def sum(input)
  array = []
  1.upto(input) {|num| array << num}
  array.sum
end

def product(input)
  array = []
  1.upto(input) {|num| array << num}
  array.inject(:*)
end

integer = ''

loop do
arrows("Please enter an integer greater than 0")
integer = gets.chomp.to_i
  if integer < 0
    puts "Invalid integer"
  else
    break
  end
end


loop do
  arrows("Enter 's' to compute the sum and 'p' to compute the product:")
  operator = gets.chomp.upcase
    if operator == 'S'
      arrows("The sum of the integers between 1 and #{integer} is #{sum(integer)}")
      break
    elsif operator == 'P'
      arrows("The sum of the integers between 1 and #{integer} is #{product(integer)}")
      break
    else
      puts "Invalid input"
    end
 end
  

File: JS210/exersies/easy1/sum_product.js
let rlSync = require('readline-sync');

let integer = Number(rlSync.question('Please enter an integer greater then 0: '));
let operator = rlSync.question("Please enter a 's' for sum or 'p' for product.").toLowerCase();

let numbers = [];

for (let counter = 1; counter <= integer; counter++) {
  numbers.push(counter);
}

console.log(numbers);

if (operator === 's') {
  let sum = numbers.reduce((acc, num) => acc + num, 0);
  console.log(`The sum of the integers between 1 and ${integer} is ${sum}`);
} else if (operator === 'p') {
  let product = numbers.reduce((acc, num) => acc * num, 1);
  console.log(`The product of the integers between 1 and ${integer} is ${product}`);